sunbird_service_api_test/src/test/java/org/sunbird/integration/test/location/LocationTest.java

```diff
@@ -0,0 +1,368 @@
+import org.sunbird.common.util.Constants;
+import org.sunbird.integration.test.common.BaseCitrusTest;
+import org.sunbird.integration.test.user.EndpointConfig.TestGlobalProperty;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+/**
+ * Class contains functional test cases for location APIs.
+ * @author arvind.
+ */
+public class LocationTest extends BaseCitrusTest {
+
+
+  private static String stateLocationId = null;
+
+  private static String districtLocationId = null;
+  private static final String CREATE_LOCATION_URI = "/v1/location/create";
+  private static final String UPDATE_LOCATION_URI = "/v1/location/update";
+  private static final String DELETE_LOCATION_URI = "/v1/location/delete";
+  private static final String LOCATION_TEMPLATE_PATH = "templates/location/create/";
+  private static final String LOCATION_TEMPLATE_PATH_UPDATE = "templates/location/update/";
+  private static final String LOCATION_TEMPLATE_PATH_DELETE = "templates/location/delete/";
+
+  private static final String STATE_CODE= "State-02";
+  private static final String STATE_NAME= "State-0001-name";
+  private static final String DISTRICT_CODE="District-02";
+  private static final String DISTRICT_NAME="District-0001-name";
+  private static Stack<String> stack = new Stack();
+
+
+  @Autowired
+  private HttpClient restTestClient;
+  @Autowired
+  private TestGlobalProperty initGlobalValues;
+  private ObjectMapper objectMapper = new ObjectMapper();
+
+  @DataProvider(name = "createStateLocationDynamicDataProvider")
+  public Object [] [] createStateLocationDynamicDataProvider () {
+    return new Object [] [] {
+        new Object[] { createStateLocationMap(), LOCATION_TEMPLATE_PATH +"createLocationSuccessResponse.json", "createState" },
+        new Object[] { createStateLocationMap(), LOCATION_TEMPLATE_PATH +"createLocationFailureResponseForDuplicateCode.json", "createLocationStateWithDupicateCode" },
+    };
+  }
+
+  @DataProvider(name = "updateStateLocationDynamicDataProvider")
+  public Object [] [] updateStateLocationDynamicDataProvider () {
+    return new Object [] [] {
+        new Object[] { updateStateLocationMap(), LOCATION_TEMPLATE_PATH_UPDATE+"update_location_name_success_response.json", "updateLocationNameForState"},
+        new Object[] { updateStateLocationTypeMap(), LOCATION_TEMPLATE_PATH_UPDATE+"update_location_type_failure_response.json", "updateLocationNameForState"},
+
+    };
+  }
+
+  @DataProvider(name = "createDistrictLocationDynamicDataProvider")
+  public Object [] [] createDistrictLocationDynamicDataProvider () {
+    return new Object [] [] {
+
+        new Object[] { createDistrictLocationMap(), LOCATION_TEMPLATE_PATH +"createLocationSuccessResponse.json", "createLocationDistrict" },
+        new Object[] { createDistrictLocationMap(), LOCATION_TEMPLATE_PATH +"createLocationFailureResponseForDuplicateCode.json", "createLocationDistrictWithDupicateCode" },
```

Comment:
typo : createLocationDistrictWithDupicateCode  "createLocationDistrictWithDuplicateCode"
make the name as "createDistrictWithDuplicateCode"